packages/watcher/README.md

```diff
@@ -0,0 +1,24 @@
+# @eth-optimism/watcher
+
+#### Watcher
+Our `Watcher` allows you to retrieve all transaction hashes related to cross domain messages such as deposits and withdrawals. In order to use, first send a transaction which sends a cross domain message, for example a deposit from L1 into L2. After sending the deposit transaction and storing the transaction hash, use `getMessageHashesFromL1Tx(l1TxHash)` to get an array of the message hashes of all of the L1->L2 messages that were sent inside of that L1 tx (This will usually just be a single element array, but it can return multiple if one L1 transaction triggers multiple deposits). `getMessageHashesFromL2Tx(l2TxHash)` does the same for L2->L1 messages. `getL2TransactionReceipt(messageHash)` takes in an L1->L2 message hash and then after 2-5 minutes, returns the receipt of the L2 tx that the message ends up getting relayed in. `getL1TransactionReceipt(messageHash)` does the same for L2->L1 messages, except the delay is 7 days.
```

Comment:
There is still a large inferential gap here, ie `In order to use, first send a transaction which sends a cross domain message, for example a deposit from L1 into L2.` isn't technical enough for somebody to just pick up and use. I think specifying `SentMessage(bytes)` event emitted on the xdomain contract would be useful